.github/workflows/dev-ci.yml

```diff
@@ -0,0 +1,65 @@
+name: Dev CI
+
+on:
+  push:
+    branches:
+      # on all branches except main where full build will be run
+      - '*'
+      - '!main'
+
+jobs:
+  build:
+    name: Build wheel
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```

Comment:
Could you use latest version for each action used? Plus pin the dependency version with the corresponding hash, keeping the version as a comment, like we did in [powsybl-core CI](https://github.com/powsybl/powsybl-core/blob/4d431b1e0c157c05b41fd549ec844bbfc638cfe4/.github/workflows/maven.yml#L25). This is for security reasons (see https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies for more details).